### PR TITLE
(GH-18) Add docker backend status method

### LINKED ISSUE #18: Scaffold DockerBackend
Implement the following:

- [x] `status` function that can verify Docker is running and available (returns a `BackendStatus` object)
- [x] Stub other methods required by `BackendI`

### PR DESCRIPTION
**Pull request by michaeltlombardi**
Prior to this PR, the docker backend was scaffolded but unimplemented.

This PR adds the new `BackendStatus` struct, which generically reports whether a particular backend is available and relevent information about that backend; in the case of an unavailable backend, it should report *why* that backend is unavailable (surfacing any error messages) and in the case of an available backend, useful information.

This PR adds a new `DockerClient` interface to the docker backend in order to enable mocking/unit testing. The only function currently used from the client is the `ServerVersion()` function, consumed in the new `Status()` function for the backend.

The `Status()` function is implemented on the docker backend to call the `ServerVersion()` docker client function. If this call errors, then the docker service/backend is unavailable. The most likely reason for this is the Docker Engine not running, so special handling is added for that edge case to handle it in a slightly more friendly way for end users.

If the `ServerVersion()` docker client function does *not* error it returns information about Docker which is useful to surface; in this case, the platform (Docker community/enterprise, desktop/server), the version, and the API version being used.

Finally, this PR adds unit testing for the new `Status()` function and includes mocks for the Docker Client. In this PR the mocks are kept in the same file as the unit tests; if/when the mocks need to be used in another test file, it may make sense to move them into a mocks folder.

Resolves #18.

### CODE REVIEW

**Review by sanfrancrisko — APPROVED** [review]

File: pkg/prm/backends/docker.go [hunk 1]
```diff
@@ -19,3 +33,33 @@ func (*Docker) Exec(tool *prm.Tool, args []string) (prm.ToolExitCode, error) {
...
+
+// Check to see if the Docker runtime is available:
+// if so, return true and info about Docker on this node;
+// if not, return false and the error message
+func (d *Docker) Status() prm.BackendStatus {
+	// The client does not error on creation if the background service is not running,
+	// but attempting to list the containers does.
+	dockerInfo, err := d.Client.ServerVersion(context.Background())
+	if err != nil {
+		message := fmt.Sprintf("%s", err)
```
Comment (da-ar) [inline pkg/prm/backends/docker.go:45]:
```suggestion
		message := err.Error()
```

File: pkg/prm/backends/docker.go [hunk 2]
```diff
@@ -19,3 +33,33 @@ func (*Docker) Exec(tool *prm.Tool, args []string) (prm.ToolExitCode, error) {
...
+		// This is 90% likely the reason this command fails;
+		// the alternative error message is lengthy and includes
+		// references to pipes and the API which are more likely
+		// to confuse than help; so trim it to the most useful info.
+		daemonNotRunning, _ := regexp.Compile(`error during connect: This error may indicate that the docker daemon is not running.`)
+		notRunningMessage := daemonNotRunning.FindString(message)
+		if notRunningMessage != "" {
```
Comment (da-ar) [inline pkg/prm/backends/docker.go:52]: Could this be simplified?

```suggestion
		daemonNotRunning := "error during connect: This error may indicate that the docker daemon is not running."
		if strings.Contains(message, daemonNotRunning) {
			message = daemonNotRunning
```

File: pkg/prm/backends/docker_test.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
+package backends
```
Comment (da-ar) [inline pkg/prm/backends/docker_test.go:1]: move tests out of the package they are testing so: `backends_test`
This ensures that the package works as a reusable component

**Review by da-ar — CHANGES_REQUESTED** [review]
Looks good. Weird that the package is docker/docker and not moby/moby?

File: pkg/prm/backends/docker.go (hunk 1 above)
Comment (sanfrancrisko) [reply pkg/prm/backends/docker.go:45]: I would even suggest `fmt.Errorf` - I think some syntax checks might actually highlight this in `pdkgo`

**Pull request merged**